spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-yarn/src/main/java/org/springframework/cloud/data/module/deployer/yarn/YarnCloudAppService.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.module.deployer.yarn;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Interface used to glue a state machine and yarn application logic together.
+ *
+ * @author Janne Valkealahti
+ */
+public interface YarnCloudAppService {
+
+	/**
+	 * Get applications pushed to hdfs.
+	 *
+	 * @return the applications
+	 */
+	Collection<CloudAppInfo> getApplications();
+
+	/**
+	 * Get running application instances.
+	 *
+	 * @return the instances
+	 */
+	Collection<CloudAppInstanceInfo> getInstances();
+
+	/**
+	 * Push new application into hdfs.
+	 *
+	 * @param appVersion the app version
+	 */
+	void pushApplication(String appVersion);
+
+	/**
+	 * Submit new application instance into yarn.
+	 *
+	 * @param appVersion the app version
+	 * @return the application id
+	 */
+	String submitApplication(String appVersion);
```

Comment:
Should the JavaDoc (either at the individual method or at the interface level) indicate the difference between "submit" and "push"? Should it describe general expected usage/lifecycle?